frontend/src/components/users/PasswordResetForm.js

```diff
@@ -0,0 +1,157 @@
+import ThingahaFormModal from '../common/ThingahaFormModal'
+import MenuItem from '@material-ui/core/MenuItem'
+import ThingahaAddressFields from '../common/ThingahaAddressFields'
+import ThingahaSelect from '../common/ThingahaSelect'
+
+const FormContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: space-between;
+  min-width: 24rem;
+
+  & .address {
+    font-weight: bold;
+    font-size: 1rem;
+  }
+
+  & .icon {
+    color: ${(props) => props.theme.palette.common.grey};
+  }
+`
+
+const StyledFormControl = styled(FormControl)`
+  width: 100%;
+  margin-bottom: 1rem;
+`
+const AddressContainer = styled.div`
+  width: 100%;
+  margin-top: 1rem;
+  margin-bottom: 0.8rem;
+`
+
+const PasswordResetForm = ({
+  visible,
+  setVisible,
+  passwordReset,
+  values,
+  setFieldValue,
+  setValues,
+  handleChange,
+  errors,
+  validateForm,
+  submitForm,
+  handleSubmit,
+}) => {
+  return (
+    <ThingahaFormModal
+      title={passwordReset ? 'Password Reset' : null }
+      open={visible}
+      onClose={() => setVisible(false)}
+      onCancel={() => setVisible(false)}
+      onSubmit={(e) => {
+        submitForm(e)
+      }}
+    >
+      <form onSubmit={handleSubmit}>
+        <FormContainer>
+          {/* <StyledFormControl>
+            <TextField
+              id="user_id"
+              name="user_id"
+              placeholder="User Id"
+              label="User Id"
+              type="text"
+              onChange={handleChange}
+              value={values.user_id}
+            />
+          </StyledFormControl> */}
+          <StyledFormControl>
+            <TextField
+              id="password"
+              name="password"
+              placeholder="********"
+              label="New Password"
+              type="password"
+              onChange={handleChange}
+              value={values.password}
+            />
+          </StyledFormControl>
+        </FormContainer>
+      </form>
+    </ThingahaFormModal>
+  )
+}
+
+const transformPasswordResetSchemaFlat = (user) => {
+  return {
+    user_id: user.id,
+    password: user.password,
+  }
+}
+
+const transformPasswordResetSchema = (user) => {
+  return {
+    user_id: user.id,
+    password: user.password,
+  }
+}
+
+const mapStateToProps = (state) => ({
+  users: state.users,
+  error: state.error,
+})
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    submitPasswordResetForm: (values) => {
+      dispatch(actions.submitPasswordResetForm(transformPasswordResetSchema(values)))
+    },
+  }
+}
+
+const FormikPasswordResetForm = withFormik({
+  mapPropsToValues: (props) => {
+    return transformPasswordResetSchemaFlat(
+      props.passwordReset || {
+        user_id: '',
+        password: '',
+      }
+    )
+  },
+
+  handleSubmit: (values, { props }) => {
+    if (props.passwordReset) {
+      console.log(values)
+      props.submitPasswordResetForm(values)
```

Comment:
Bro, the error is happening because you need to pass in user id using the `passwordReset` props that you passed in.

eg.
```javascript
props.submitPasswordResetForm({
    password: values.password,
    id: props.passwordReset.id,
})
```
You already have the user id from the props so we don't need to get it from the form `values` .